server/Dockerfile

```diff
@@ -34,15 +34,13 @@ RUN apt-get update ;\
 USER appuser
```

Comment:
I'm commenting here so there's an easy thread.

I'm not a huge fan of this implementation. Look at all of the files and interfaces you had to change to make it, and how intrusive this change is. All of it in order to just make a small that was previously external and backend agnostic.

You now have to implement it for every new backend we add for cache, queue, and whatever, and the API is uglier for it.

I can't help but think that having literally just this in a function that's called from `main` would have been better:
```rust
for i in 1..=times_to_retry {
    if let Ok(_) = tcp_try_connect(hostname, port).await {
        break;
    }
    sleep(backoff_sleep).await;
}
```